### ISSUE TITLE
Default Color Schemes should have accessible contrast

### ISSUE THREAD

**Issue by mathetos** [WordPress/twentysixteen#177]
For web accessibility, there should be more contrast in the color schemes. Here's a good resource on that: http://webaim.org/resources/contrastchecker/

These schemes specifically fail on the link colors against their background colors:
## GREEN

![webaim_colorcontrastchecker](https://cloud.githubusercontent.com/assets/930724/9862515/4ad98e0e-5aee-11e5-9986-5f841eb6ab91.png)
## GRAY

![webaim_colorcontrastchecker-gray](https://cloud.githubusercontent.com/assets/930724/9862595/a66e2ab8-5aee-11e5-89a1-fbe4afbb3df8.png)

**Comment by mathetos** [comment]
I see that this was discussed here https://github.com/WordPress/twentysixteen/issues/133 but the issue of Accessibility wasn't raised in that conversation.

I think accessibility should be a priority, not just simply a design decision.

**Comment by iamtakashi** [comment]
The Green and the Gray color schemes are not default, they are options. The default is "Default" color scheme.

> The default settings will be the only color scheme checked. If a theme offers multiple color schemes, only the default scheme is required to pass these guidelines.

https://make.wordpress.org/themes/handbook/review/accessibility/required/#contrasts

**Comment by cgrymala** [comment]
So, @iamtakashi, what you're saying is that, instead of being an example for good design standards and what should be done with WordPress, you want TwentySixteen to be an example of the minimum that can be done to pass the theme review?

As a color-blind WordPress user, I have to say that I am extremely offended by that attitude, and I implore you to re-open this ticket and fix these color schemes.

I would be upset if we were talking about admin color schemes, but at least, as a user of the system, I'd have a choice in which color scheme is used. However, in this case, we're talking about front-end color schemes, and it will be a serious black mark against WordPress itself if the default theme ends up being shipped with any inaccessible color schemes included.

As an end-user (a visitor of any site using TwentySixteen), I have no real control over the color scheme that the site manager chooses, and, as such, I am left with little other option than to abandon the site.

Again, this would be an extremely poor representation of WordPress itself, and I am very sad that you take such a lax attitude toward those of us that can't read a site because of your poor design decisions.

Also, you left out an important note from the guidelines you quoted:

> Alternative themes should be clearly labeled if they do not meet accessibility guidelines.

I assume, then, that Twenty Sixteen will include a giant warning telling people not to use any of the alternative color schemes, because nearly 10% of the world's population won't be able to read links on sites using the scheme, right?

**Comment by cgrymala** [comment]
As a side note, maybe your quote from the theme standards means that the theme standards themselves should be updated, rather than theme developers being able to use them as a cop-out.

**Comment by iamtakashi** [comment]
@cgrymala, Three out of five colour schemes are high contrast — the default, dark, and yellow schemes (although the link colour in the yellow is a bit low contrast). So, I disagree with you to call it as an example of the minimum. Also all colours are editable in any of colour scheme, the theme doesn't force to use any of the preset of colour.

It's true that we don't have control over what colour scheme or what colour for each element site owner choose. But, whatever the choice is, there most likely is a good reason for them to choose.

Having two low contrast colour schemes, calling it "an extremely poor representation of WordPress itself" is a bit of stretch. And I do not agree with the idea of all colour schemes need to be high contrast.

**Comment by cgrymala** [comment]
So, what you're saying is that the developers of TwentySixteen, and, by extension, WordPress, are only 60% committed to accessibility? Granted, it's better than a lot of groups, but it's not what I'd expect from WP. 

There is no good reason to ignore accessibility standards when you have the option to do the right thing. If users tweak their own color settings without recognizing what they're doing, that's one thing. If you provide them with a poor design that explicitly excludes visitors from being able to consume content, that's a different matter altogether. 

Would we be having this same sort of discussion if there was a ticket making you aware that only 3/5 of the text in TwentySixteen was translatable?

**Comment by cgrymala** [comment]
I don't know that I'd necessarily qualify even the default theme as "high contrast" in all areas. It may meet WCAG 2.0 AA contrast guidelines, but there are quite a few areas that don't meet the AAA guidelines (4.5:1 for large text, 7:1 for body text).

I'm not saying they necessarily _should_ meet AAA guidelines, yet, but if they don't, they certainly don't qualify as "high contrast".

The standard link color (`#007acc`) has a contrast of 4.51:1, which just barely passes AA level and is a far cry from AAA guidelines. The hover link color (`#757575`) has a contrast of 4.6:1, which, again, is a far cry from AAA guidelines.

The default color scheme does, just barely, pass AA guidelines, but it is certainly not a "high contrast" theme, as much as you may like it to be.

The "Dark" color scheme would most likely qualify under anyone's definition of high contrast.

As you mention, the link colors (hover in the body; standard in the sidebars) in the _Yellow_ color scheme definitely do not have enough contrast. The color (`#7f7d6f`) has a contrast of 3.55:1. You could use `#6f6d61` in order to get to 4.5:1, at least.

_Regarding this original ticket:_

**Gray
You could easily change `#aaaaaa` to `#cacaca` to at least reach 4.5:1 contrast

**Green
The white text on the green background is virtually unreadable. Following are 2 possible configurations that would bring this Green color scheme into AA contrast compliance.
- Main body text from `#ffffff` to `#000000`
- Link color from `#6d8c87` to `#484848`
- Secondary color from `#efeef4` to `#4d476a`

Or:
- Background color from `#acc1a2` to `#4f6745`
- Link color from `#6d8c87` to `#d4dedc`

**Comment by TheHeat** [comment]
I agree that if these colour schemes are being offered to site-owners, many of whom will have no knowledge of a11y standards at all, they should _all_ comply with colour contrast standards.

It is not unreasonable to expect _everything_ that comes out of a high profile project like WordPress to meet these standards.
